lib/scholar/neighbors/kd_tree.ex

```diff
@@ -462,24 +450,36 @@ defmodule Scholar.Neighbors.KDTree do
     end
   end
 
-  defnp query_one_point(point, tree, opts) do
+  defnp query_points(point, tree, opts) do
     k = opts[:k]
     node = Nx.as_type(root(), :s64)
+    point = point |> Nx.vectorize(:x)
```

Comment:
Take a look in NxSignal. There's a bit of a treatment you could apply using revectorize to allow for the input to be vectorized beforehand as well